go/options/command_options.go

```diff
@@ -156,22 +157,70 @@ func (opts *GetExOptions) ToArgs() ([]string, error) {
 
 // Expiry is used to configure the lifetime of a value.
 type Expiry struct {
-	Type  constants.ExpiryType
-	Count uint64
+	Type      constants.ExpiryType
+	Duration  uint64
+	Timestamp time.Time
 }
 
-func NewExpiry() *Expiry {
-	return &Expiry{}
+// isExpiryTypeSeconds checks if the expiry type should be in seconds
+func isExpiryTypeSeconds(duration time.Duration) bool {
+	return duration%time.Second == 0
 }
 
+// NewExpiryIn creates a new Expiry with a duration from now
+func NewExpiryIn(duration time.Duration) *Expiry {
+	dur := int(duration.Milliseconds())
+	expiryType := constants.Milliseconds
+	if isExpiryTypeSeconds(duration) {
+		expiryType = constants.Seconds
+		dur = int(duration.Seconds())
+	}
+	return &Expiry{
+		Type:     expiryType,
+		Duration: uint64(dur),
+	}
+}
+
+// NewExpireAt creates a new Expiry with a specific timestamp
+func NewExpireAt(timestamp time.Time) *Expiry {
```

Comment:
Let's keep it consistent
```suggestion
// NewExpiryAt creates a new Expiry with a specific timestamp
func NewExpiryAt(timestamp time.Time) *Expiry {
```